Added the guidlines to enable 'start_event_grains' config (#88)
uyuni-project/uyuni-docs: modules/client-configuration/pages/tshoot-clients.adoc

> == Passing the grains to the start event

Reviewer: ```suggestion
== Passing Grains to a Start Event
```

> On each minion start up, in order to identify if minion is already registered or not, ``machine_id`` grain is needed. And in order to get this grain

Reviewer: ```suggestion
Every time a Salt client starts, it passes the ``machine_id`` grain to {productname}.
{productname} uses this grain to determine if the client is registered.
```

> a synchronous salt call was needed. Which was expensive because of obvious blocking nature of synchronous calls, it was even more noticeable when a lot of minions

Reviewer: ```suggestion
This process requires a synchronous Salt call.
Synchronous Salt calls block other processes, so if you have a lot of clients start at the same time, the process could create significant delays.
```

> were being started at the same time.

Reviewer: ```suggestion
```

> To overcome this problem, a new feature has been introduced in upstream salt, user can list the minion grains that should be passed in the payload of the start event and salt

Reviewer: ```suggestion
To overcome this problem, a new feature has been introduced in Salt to avoid making a separate synchronous Salt call.
```

> will take care of the rest. Now there is no need to make a separate synchronous call.

Reviewer: ```suggestion
```

> If user add this config to the minion and minion supports this feature, a significant performance gain can be achieved. But to add this config to all the existing minions is

Reviewer: ```suggestion
To use this feature, you can add a configuration parameter to the client configuration, on clients that support it.
```

> not so trivial. User can use provided `mgr_start_event_grains.sls` helper salt state with the following guidelines to make this process safe and relatively less painful.

Reviewer: ```suggestion
To make this process easier, you can use the ``mgr_start_event_grains.sls`` helper Salt state.
```

[NOTE]
====

> This only applies to older Salt clients.

Reviewer: Older than ... ?

If you registered your Salt client recently, this config parameter is added by default.
====

> On the {productname} Server, at the command prompt, use this command to enable the ``start_event_grains`` config :

Reviewer: ```suggestion
On the {productname} Server, at the command prompt, use this command to enable the ``start_event_grains`` configuration helper:
```

----
salt '*' state.sls util.mgr_start_event_grains
----

> This command will add the required configuration into the minion config file and will get into effect whenever minion will get restarted.

Reviewer: ```suggestion
This command adds the required configuration into the client's configuration file, and applies it when the client is restarted.
```
